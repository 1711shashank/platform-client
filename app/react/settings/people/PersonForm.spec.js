import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import PersonForm from "./PersonForm";

test("a Person Form with roles displays roles correctly", () => {
    const saveNewPerson = () => {};
    const roles = [{ id: "email", name: "email" }];
    const isLoadingRoles = false;
    const component = renderer.create(
        <PersonForm
            saveNewPerson={saveNewPerson}
            roles={roles}
            isLoadingRoles={isLoadingRoles}
        />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test("A Person Form without roles displays loading instead", () => {
    const saveNewPerson = () => {};
    const roles = [];
    const isLoadingRoles = true;
    const component = renderer.create(
        <PersonForm
            saveNewPerson={saveNewPerson}
            roles={roles}
            isLoadingRoles={isLoadingRoles}
        />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test("the handleChange method updates the input value", () => {
    const saveNewPerson = () => {};
    const roles = [{ id: "email", name: "email" }];
    const isLoadingRoles = false;

    const personForm = shallow(
        <PersonForm
            saveNewPerson={saveNewPerson}
            roles={roles}
            isLoadingRoles={isLoadingRoles}
        />
    );

    expect(personForm.find('[id="realname"]').props().value).toBe("");
    const realname = personForm.find('[id="realname"]');
    realname.simulate("change", {
        target: { value: "My real name", id: "realname" }
    });
    expect(personForm.find('[id="realname"]').props().value).toBe("My real name");
});

// test for submit success

// test for submit error
