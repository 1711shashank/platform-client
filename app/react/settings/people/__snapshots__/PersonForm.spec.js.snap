// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A Person Form without roles displays loading instead 1`] = `
<main
  role="main"
>
  <div>
    <h3>
      Add people to Ushahidi
    </h3>
    <p>
      Add members of your team, stakeholders, and other members of your community to Ushahidi.
    </p>
  </div>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="realname"
    >
      Name
      <input
        id="realname"
        onChange={[Function]}
        placeholder="What is this person's full name"
        required={true}
        type="text"
        value=""
      />
    </label>
    <label
      htmlFor="email"
    >
      Email
      <input
        id="email"
        onChange={[Function]}
        placeholder="email"
        required={true}
        type="text"
        value=""
      />
    </label>
    <label
      htmlFor="password"
    >
      Password
      <input
        id="password"
        onChange={[Function]}
        placeholder="password"
        required={true}
        type="password"
        value=""
      />
    </label>
    <p>
      Loading...
    </p>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</main>
`;

exports[`a Person Form with roles displays roles correctly 1`] = `
<main
  role="main"
>
  <div>
    <h3>
      Add people to Ushahidi
    </h3>
    <p>
      Add members of your team, stakeholders, and other members of your community to Ushahidi.
    </p>
  </div>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="realname"
    >
      Name
      <input
        id="realname"
        onChange={[Function]}
        placeholder="What is this person's full name"
        required={true}
        type="text"
        value=""
      />
    </label>
    <label
      htmlFor="email"
    >
      Email
      <input
        id="email"
        onChange={[Function]}
        placeholder="email"
        required={true}
        type="text"
        value=""
      />
    </label>
    <label
      htmlFor="password"
    >
      Password
      <input
        id="password"
        onChange={[Function]}
        placeholder="password"
        required={true}
        type="password"
        value=""
      />
    </label>
    <label
      htmlFor="role"
    >
      Role
      <select
        id="role"
        onChange={[Function]}
        value="Select a role"
      >
        <option
          id="default"
          value="Select a role"
        >
          Select a role
        </option>
        <option
          id="email"
        />
      </select>
    </label>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</main>
`;
